modules/installation-gcp-shared-vpc-config.adoc

```diff
@@ -0,0 +1,82 @@
+There are several configuration parameters which are required to install {product-title} on GCP using a shared VPC. The following is a sample `install-config.yaml` file which demonstrates these fields.
+
+[IMPORTANT]
+====
+This sample YAML file is provided for reference only. You must modify this file with the correct values for your environment and cluster.
+====
+
+[source,yaml]
+----
+apiVersion: v1
+baseDomain: example.com
+credentialsMode: Passthrough <1>
+metadata:
+  name: cluster_name
+platform:
+  gcp:
+    computeSubnet: shared-vpc-subnet-1 <2>
+    controlPlaneSubnet: shared-vpc-subnet-2 <3>
+    createFirewallRules: Disabled <4>
+    network: shared-vpc <5>
+    networkProjectID: host-project-name <6>
+    publicDNSZone:
+      id: public-dns-zone <7>
+      project: host-project-name <8> 
+    privateDNSZone:
+      id: private-dns-zone <9>
+      project: host-project-name <10>
+    projectID: service-project-name <11>
+    region: us-east1
+    defaultMachinePlatform:
+      tags: <12>
+      - global-tag1
+controlPlane:
+  name: master
+  platform:
+    gcp:
+      tags: <12>
+      - control-plane-tag1
+      type: n2-standard-4
+      zones:
+      - us-central1-a
+      - us-central1-c
+  replicas: 3
+compute:
+- name: worker
+  platform:
+    gcp:
+      tags: <12>
+      - control-plane-tag1 
+      type: n2-standard-4
+      zones:
+      - us-central1-a
+      - us-central1-c
+  replicas: 3
+networking:
+  clusterNetwork:
+  - cidr: 10.128.0.0/14
+    hostPrefix: 23
+  machineNetwork:
+  - cidr: 10.0.0.0/16
+pullSecret: '{"auths": ...}'
+sshKey: ssh-ed25519 AAAA... <13>
+----
+<1> `credentialsMode` must be set to Passthrough to allow the cluster to use the provided GCP service account after cluster creation. See the "Prerequisites" section for the required GCP permissions that your service account must have.
+<2> The name of the subnet in the shared VPC for compute machines to use.
+<3> The name of the subnet in the shared VPC for control plane machines to use.
+<4> Optional. If you set `createFirewallRules` to "Disabled", you can create and manage firewall rules manually through the use of network tags. By default, the cluster will automatically create and manage the firewall rules that are required for cluster communication. Your service account must have `roles/compute.networkAdmin` and `roles/compute.securityAdmin` privileges in the host project to perform these tasks automatically. 
+<5> The name of the shared VPC.
+<6> The name of the host project where the shared VPC exists.
+<7> Optional. The name of a public DNS zone in the host project. Your service account must have the `roles/dns.admin` privilege in the host project. If you do not set this value, the installation program will use the public DNS zone in the service project.
+<8> Optional. The name of the host project which contains the public DNS zone. This value is required if you specify a public DNS zone that exists in another project.
```

Comment:
FYI there's a [bug](https://issues.redhat.com/browse/OCPBUGS-2351) on `platform.gcp.publicDNSZone`, so we are not sure yet when this settings is specified, what's expected for the `baseDomain`. @jstuever Would you please advise? Thanks!

Reply:
@jstuever @bscott-rh Besides, we guess that if both `platform.gcp.publicDNSZone` and `platform.gcp.privateDNSZone` are specified, their DNS names should match. Similarly, if `platform.gcp.privateDNSZone` is specified alone, its DNS name should match `baseDomain` (of the service project public DNS zone), Please correct me, thanks!

Reply:
Added a note that the public DNS zone must have the same domain as the baseDomain.